### PR TITLE
make record formatting suck less

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Bugfix
- Enhancement / new feature
- Refactoring

### Description

The maven build applies code formatting based on the Eclipse code style settings, this updates to the latest version and reformats things to match. Thus fixing the horrible formatting of records.

### Additional Context

The maven formatter plugin and check style are fighting in https://github.com/kroxylicious/kroxylicious/pull/602 so something has to change. If we are changing anything we might as well get better formatting of records too.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious/src/main/java/io/kroxylicious/proxy/internal/net/EndpointRegistry.java [hunk 1]
```diff
@@ -134,9 +135,10 @@ private record VirtualClusterRecord(CompletionStage<Endpoint> registrationStage,
             Objects.requireNonNull(deregistrationStage);
         }
 
-    private static VirtualClusterRecord create(CompletionStage<Endpoint> stage) {
-        return new VirtualClusterRecord(stage, new AtomicReference<>(), new AtomicReference<>());
-    }}
+        private static VirtualClusterRecord create(CompletionStage<Endpoint> stage) {
+            return new VirtualClusterRecord(stage, new AtomicReference<>(), new AtomicReference<>());
+        }
+    }
```
Comment (SamBarker) [inline kroxylicious/src/main/java/io/kroxylicious/proxy/internal/net/EndpointRegistry.java:141]: This is a good example of actually getting the formatting right within a record that this change brings.

**Review by SamBarker — COMMENTED** [review]
The process was, update settings in Eclipse, export those to file. Import into Intellij and have Intellij re-export to `.editorconfig` then run a full maven build to format everything.

File: integrationtests/src/test/java/io/kroxylicious/proxy/TlsIT.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ class TlsIT extends BaseIT {
     private static final HostPort PROXY_ADDRESS = HostPort.parse("localhost:9192");
     private static final ClusterNetworkAddressConfigProviderDefinition CONFIG_PROVIDER_DEFINITION = new ClusterNetworkAddressConfigProviderDefinitionBuilder(
             "PortPerBroker").withConfig("bootstrapAddress", PROXY_ADDRESS)
-            .build();
+                    .build();
```
Comment (robobario) [inline integrationtests/src/test/java/io/kroxylicious/proxy/TlsIT.java:56]: this block looks a bit weird:
```
private static final ClusterNetworkAddressConfigProviderDefinition CONFIG_PROVIDER_DEFINITION = new ClusterNetworkAddressConfigProviderDefinitionBuilder(
            "PortPerBroker").withConfig("bootstrapAddress", PROXY_ADDRESS)
                    .build();
```

Comment (SamBarker) [reply integrationtests/src/test/java/io/kroxylicious/proxy/TlsIT.java:56]: hmm, yeah. Tweaking it also leads to a fight between intellij and Maven.... Will have to open Eclipse again to see if I can get it right there.

**Review by robobario — APPROVED** [review]
LGTM with nits, will be great having inners record indented properly

File: integrationtests/src/test/java/io/kroxylicious/proxy/TlsIT.java (hunk 2 above)
Comment (SamBarker) [reply integrationtests/src/test/java/io/kroxylicious/proxy/TlsIT.java]: Not sure why this is being changed, but we have a line length limit of 170 (this hasn't changed) which is reached somewhere in the string `PortPerBroker`. There are no good answers  for this one.

The best I've managed is 
```
 private static final ClusterNetworkAddressConfigProviderDefinition CONFIG_PROVIDER_DEFINITION = new ClusterNetworkAddressConfigProviderDefinitionBuilder(
            "PortPerBroker")
            .withConfig("bootstrapAddress", PROXY_ADDRESS)
            .build();
```

**Pull request merged**